Docs for Custom Theme feature support in Designer (#281)
vaadin/docs: articles/tools/designer/using-designer/theming.asciidoc

[[figure.designer.designing.project.theme]]
.Theme settings
image::images/project-theme-change.png[]

[role="since:com.vaadin:vaadin@V19"]
= Custom Theme

> Vaadin Designer supports theming the same way as Flow (see <<{articles}/ds/customization/custom-theme, Custom Theme>> for more information).

Reviewer: Cross references need be appended with `#`, otherwise they are interpreted as in-page references.
```suggestion
Vaadin Designer supports theming the same way as Flow (see <<{articles}/ds/customization/custom-theme#, Custom Theme>> for more information).
```

> Vaadin Designer supports theming the same way as Flow (see <<{articles}/theming/creating-a-custom-theme, Custom Theme>> for more information).

Reviewer: Outdated cross reference/link. This has been moved under design system / customization.

Author: Fixed.

When a design is opened, Vaadin Designer:

>  * Modifies `package.json` file to add utility libraries. These changes should be commited to your version control system.

Reviewer: ```suggestion
 * Modifies `package.json` file to add utility libraries. These changes should be committed to your version control system.
```

 * Then installs frontend dependencies if they are not available in the project. This installation process could take a long time because the IDE also needs to re-index the new files, but this just happens once.
 * Loads the custom theme, defined by the `@Theme` annotation
 * Loads all style sheets of the theme (and its parent themes if there is any), such as global and component style sheets.

>  * Static resources (images, fonts, ...) of the currently using theme are resolved.

Reviewer: ```suggestion
 * Static resources (images, fonts, ...) of the theme are resolved.
```

When a style sheet under a custom theme is changed, Designer is automatically reloaded to apply the changes.

[NOTE]

> When switching the theme under @Theme, all Designer visual editors need to be re-opened to apply the new theme. Also in Preview mode, the browser isn't auto reloaded when styles are changed, you need to refresh the browser manually.

Reviewer: ```suggestion
When updating the `@Theme` annotation value, all Designer visual editors need to be re-opened to apply the new theme. Also in Preview mode, you need to refresh the browser manually.
```

> When switching the theme under @Theme, all Designer visual editors need to be re-opened to apply the new theme. Also in Preview mode, the browser isn't auto reloaded when styles are changed, you will need to refresh the browser manually.

Reviewer: ```suggestion
When switching the theme under @Theme, all Designer visual editors need to be re-opened to apply the new theme. Also in Preview mode, the browser isn't auto reloaded when styles are changed, you need to refresh the browser manually.
```
